tests/components/enphase_envoy/test_config_flow.py

```diff
@@ -361,3 +363,55 @@ async def test_reauth(hass: HomeAssistant, config_entry, setup_enphase_envoy) ->
     )
     assert result2["type"] == "abort"
     assert result2["reason"] == "reauth_successful"
+
+
+async def test_phase_data(hass: HomeAssistant, setup_enphase_envoy) -> None:
+    """Test we can setup from zeroconf and get data."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data=zeroconf.ZeroconfServiceInfo(
+            ip_address=ip_address("1.1.1.1"),
+            ip_addresses=[ip_address("1.1.1.1")],
+            hostname="mock_hostname",
+            name="mock_name",
+            port=None,
+            properties={"serialnum": "1234", "protovers": "7.0.0"},
+            type="mock_type",
+        ),
+    )
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            "host": "1.1.1.1",
+            "username": "test-username",
+            "password": "test-password",
+        },
+    )
+    assert result2["result"].unique_id == "1234"
+
+    envoy = hass.data[DOMAIN][result2["result"].entry_id].envoy
```

Comment:
We should not access internals during tests.

I am not up to date on the test state of the integration, but these ideally would be sensor tests as in 
1. We set up the integration with this as source data
2. Observe the created states and check if they are what we expect
3. profit

Reply:
I'll change to use   state = hass.states.get("sensor.whatever")

Reply:
May require more overhaul of current test. Currently the device is set up, but no entities, so no states. The flexit_bacnet example does this, but is setup different as enphase currently.

Reply:
While it would be nice to have these tests, its better for them to be done in another PR where we create a `test_sensor.py`

Reply:
Ok, I'll revert the test change. With the new CT data in the 1.18 library version there will be another HA PR for net-consumption and net-production entities to use with Energy dashboard that need sensor testing as well. So a good foundation for that would be neat.